src/OrbitGl/CaptureWindow.cpp

```diff
@@ -320,43 +333,40 @@ void CaptureWindow::KeyPressed(unsigned int key_code, bool ctrl, bool shift, boo
       ToggleRecording();
       break;
     case 18:  // Left
-      if (time_graph_ == nullptr) return;
-      if (shift) {
-        time_graph_->JumpToNeighborTimer(app_->selected_timer(),
-                                         TimeGraph::JumpDirection::kPrevious,
+      if (time_graph_ == nullptr || !IteratableType(timer->type())) return;
+      if (shift && FunctionIteratableType(timer->type())) {
+        time_graph_->JumpToNeighborTimer(timer, TimeGraph::JumpDirection::kPrevious,
                                          TimeGraph::JumpScope::kSameFunction);
-      } else if (alt) {
-        time_graph_->JumpToNeighborTimer(app_->selected_timer(),
-                                         TimeGraph::JumpDirection::kPrevious,
+      } else if (alt && FunctionIteratableType(timer->type())) {
+        time_graph_->JumpToNeighborTimer(timer, TimeGraph::JumpDirection::kPrevious,
                                          TimeGraph::JumpScope::kSameThreadSameFunction);
       } else {
-        time_graph_->JumpToNeighborTimer(app_->selected_timer(),
-                                         TimeGraph::JumpDirection::kPrevious,
+        time_graph_->JumpToNeighborTimer(timer, TimeGraph::JumpDirection::kPrevious,
                                          TimeGraph::JumpScope::kSameDepth);
       }
```

Comment:
Yeah...I've been thinking about it a lot. It looks cleaner than it was with your other comments, but this way, when you're on GPU tracks or other tracks where you can iterate through timers and not functions, when you click shift/alt+arrow key, it will still iterate over tracks rather than do nothing at all. To get this functionality in JumpToNeighborTimer, I'd basically be checking the JumpScope && the type and sometimes changing the JumpScope back to kSameDepth.  Or just not do anything when it doesn't make sense to iterate over functions for a timer. OnLeft/Right methods don't have enough information to make this logic (OnLeft/Right change based on JumpScope). 

What do you think?

I'll also be adding the tracks that make sense to iterate over and aren't on there yet and thinking about making shift/alt iterate by gpu stage if it's a gpu track, which will either mean another JumpScope with more logic here or overload kSameFunction/kSameThreadSameFunction JumpScopes in JumpToNeighborTimer.

Reply:
The thing that is weird is that we're creating those local variables for every key press, even when we don't have to. We could have a special ArrowKeyPressed(...) method that would be called from KeyPressed(...), we'd have to do the key dispatch twice, but I think it's cleaner than creating the local variables for all keys instead of only when needed. Something like:

```
case 18:  // Left
case 20:  // Right
case 19:  // Up
case 21:  // Down
  OnArrowKeyPressed(key_code, ctrl, shift, alt);
break;
```

If we were to do that we should probably take the opportunity to define constants for the keys, (kLeftArrow instead of 18, etc.) 

WDYT?

Reply:
I think that makes sense and I now I agree with your first idea :) See if that looks better to you.